yarn-project/validator-client/src/validator.ts

```diff
@@ -207,30 +219,32 @@ export class ValidatorClient extends WithTracer implements Validator {
     };
     this.log.verbose(`Received request to attest for slot ${slotNumber}`);
 
-    // Check that I am in the committee
-    if (!(await this.epochCache.isInCommittee(this.keyStore.getAddress()))) {
-      this.log.verbose(`Not in the committee, skipping attestation`);
+    // Check that I have any address in current committee
+    const inCommittee = await this.epochCache.filterInCommittee(this.keyStore.getAddresses());
+    // if (!(await this.epochCache.isInCommittee(this.keyStore.getAddress()))) {
+    if (inCommittee.length === 0) {
+      this.log.verbose(`No validator in the committee, skipping attestation`);
       return undefined;
     }
 
     // Check that the proposal is from the current proposer, or the next proposer.
     const invalidProposal = await this.blockProposalValidator.validate(proposal);
     if (invalidProposal) {
       this.log.verbose(`Proposal is not valid, skipping attestation`);
-      this.metrics.incFailedAttestations('invalid_proposal');
+      this.metrics.incFailedAttestations(inCommittee.length, 'invalid_proposal');
       return undefined;
     }
 
     // Check that the parent proposal is a block we know, otherwise reexecution would fail.
     // Q: Should we move this to the block proposal validator? If there, then p2p would check it
-    // before re-broadcasting it. This means that proposals built on top of an L1-reorgd-out block
+    // before re-broadcasting it. This means that proposals built on top of an L1-reorg'ed-out block
     // would not be rebroadcasted. But it also means that nodes that have not fully synced would
     // not rebroadcast the proposal.
     if (blockNumber > INITIAL_L2_BLOCK_NUM) {
       const parentBlock = await this.blockSource.getBlock(blockNumber - 1);
       if (parentBlock === undefined) {
         this.log.verbose(`Parent block for ${blockNumber} not found, skipping attestation`);
-        this.metrics.incFailedAttestations('parent_block_not_found');
+        this.metrics.incFailedAttestations(inCommittee.length, 'parent_block_not_found');
```

Comment:
I don't think we need to add a metric for every address, it would just be confusing I think. My node failed to attest, that's all I really need to know I think.